File: app/src/main/java/com/afoxxvi/alopex/util/FoxTools.java
package com.afoxxvi.alopex.util;

import java.time.LocalDateTime;
import java.time.ZoneOffset;
import java.util.Calendar;
import java.util.concurrent.Executors;
import java.util.concurrent.ScheduledExecutorService;
import java.util.concurrent.ScheduledThreadPoolExecutor;
import java.util.concurrent.ThreadFactory;

public class FoxTools {
    public static LocalDateTime getLocalDateTimeFromMills(long mills) {
        return LocalDateTime.ofEpochSecond(mills / 1000, (int) (mills % 1000 * 1000000),
                ZoneOffset.ofTotalSeconds(Calendar.getInstance().get(Calendar.ZONE_OFFSET) / 1000));
    }

    public static ScheduledExecutorService getSingleExecutorService() {
        return new ScheduledThreadPoolExecutor(1, new ThreadFactory() {
            final ThreadFactory factory = Executors.defaultThreadFactory();

            @Override
            public Thread newThread(Runnable r) {
                return factory.newThread(r);
            }
        });
    }
}

File: app/src/main/java/com/afoxxvi/alopex/util/FoxTools.kt
package com.afoxxvi.alopex.util

import java.time.LocalDateTime
import java.time.ZoneOffset
import java.util.*
import java.util.concurrent.Executors
import java.util.concurrent.ScheduledExecutorService
import java.util.concurrent.ScheduledThreadPoolExecutor
import java.util.concurrent.ThreadFactory

object FoxTools {
    fun getLocalDateTimeFromMills(mills: Long): LocalDateTime {
        return LocalDateTime.ofEpochSecond(
            mills / 1000, (mills % 1000 * 1000000).toInt(),
            ZoneOffset.ofTotalSeconds(Calendar.getInstance()[Calendar.ZONE_OFFSET] / 1000)
        )
    }

    val singleExecutorService: ScheduledExecutorService
        get() = ScheduledThreadPoolExecutor(1, object : ThreadFactory {
            val factory = Executors.defaultThreadFactory()
            override fun newThread(r: Runnable): Thread {
                return factory.newThread(r)
            }
        })
}